windows/src/buildtools/build_standards_data/Keyman.System.BuildLibPalasoAllTags.pas

```diff
@@ -56,22 +61,65 @@       FResult.Add('  TLibPalasoAllTagsMap = class');
       FResult.Add('class procedure TLibPalasoAllTagsMap.Fill(dict: TDictionary<string,TArray<string>>);');
       FResult.Add('begin');
 
-      for i := 0 to FAllTags.Count - 1 do
-      begin
-        // Line format: ignore '*'
-        s := FAllTags[i];
-        items := s.Split([' ','=','*'], TStringSplitOptions.ExcludeEmpty);
-        if Length(items) = 0 then
-          Continue;
-        t := Format('  dict.Add(''%s'', [', [items[0]]);
-        for j := 1 to High(items) do
+      dict := TStringList.Create; //' TArray<string,TArray<string>>.Create;
+      try
+        for i := 0 to FAllTags.Count - 1 do
         begin
-          if j > 1 then
-            t := t + ', ';
-          t := t + Format('''%s''', [items[j]]);
+          // Line format: ignore '*'
+          s := FAllTags[i];
+          items := s.Split([' ','=','*'], TStringSplitOptions.ExcludeEmpty);
+          if Length(items) = 0 then
+            Continue;
+
+          n := items[0].IndexOf('-');
+          if n >= 0 then
+          begin
+            // This is a lang-script tag. So we will
+            // strip the basic
```

Comment:
Is it possible to just have lang-region tag?
e.g. in alltags:
```
kha-IN
khb-CN
```

Maybe validate the script tag is 4 digits.

Reply:
I think we'll worry about this in a future round. There are so many edge cases...